doc: Update software maturity doc (#7599)
nrfconnect/sdk-nrf: doc/nrf/software_maturity.rst

   :local:
   :depth: 2

|NCS| supports its various features and components at different levels of software maturity.

> The tables at the end of this page summarize the maturity level for each feature and component supported in |NCS|.

Reviewer: Shouldn't NCS be preceded by `the`?

> Three levels are used in the tables to classify the software maturity of each feature and component:

Reviewer: ```suggestion
The following categories are used in the tables to classify the software maturity of each feature and component:
```
To avoid using the number in the introduction of the list.

Supported

>    The feature or component is maintained and is suitable for product development.

Reviewer: ```suggestion
   The feature or component is implemented and maintained, and is suitable for product development.
```

Not supported
   The feature or component does not work.

Experimental

>    The feature can be used for development but it is not recommended for production.

Reviewer: ```suggestion
   The feature can be used for development, but it is not recommended for production.
```

   This means that the feature is incomplete in functionality or verification and can be expected to change in future releases.

>    The feature is made available in its current state but the design and interfaces can change between release tags.

Reviewer: ```suggestion
   The feature is made available in its current state, but the design and interfaces can change between release tags.
```

   The feature is also labeled as ``EXPERIMENTAL`` in Kconfig files to indicate this status.

   .. note::
      By default, the build system generates build warnings to indicate when features labeled ``EXPERIMENTAL`` are included in builds.

>       You can disable these warnings by disabling the :kconfig:option:`CONFIG_WARN_EXPERIMENTAL` kconfig option.

Reviewer: Can we link here? https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/app_build_system.html#ncs-additions

   :local:
   :depth: 2

|NCS| supports its various features and components at different levels of software maturity.

The following are the three levels used to classify the software maturity of each feature and component:

* *Supported* - The feature or component is maintained and is suitable for product development.

> * *Not supported* - The feature or component does not work.

Reviewer 2: Mention that this is `-` in the tables?

Author: fixed

* *Experimental* - The feature can be used for development but it is not recommended for production.

>   This means that the feature is incomplete in functionality or verification and can be expected to change in future releases.

Reviewer 2: This doesn't work visually. It's really hard to see that this text is for "Experimental". Could we use a table or something more visible?

![image](https://user-images.githubusercontent.com/12450381/168816929-844c2679-ea72-4d49-982c-d81b79a06d84.png)

Author: I agree this version is not perfect, but it still looked better than before. I'll try to come up with a better idea.

Reviewer: Try using the definition list:

```suggestion
Supported
   The feature or component is maintained and is suitable for product development.

Not supported
   The feature or component does not work.

...
```

Author: Right, fixed

     - Customer issues raised for experimental features on tagged |NCS| releases are handled by technical support on DevZone.
     - Not available.
   * - **Bug fixing**

>      - Reported critical bugs will be resolved in both main and latest tagged versions.

Reviewer: ```suggestion
     - Reported critical bugs will be resolved in both ``main`` and the latest tagged versions.
```

     - Bug fixes and improvements are not guaranteed to be applied.
     - Not available.
   * - **Implementation completeness**
     - Complete implementation of the agreed features set.
     - Significant changes may be applied in upcoming versions.
     - A feature or component is available as an implementation, but is not compatible with (or tested on) a specific device or series of products.
   * - **API definition**
     - The API definition is stable.

>      - The API definition is not stable and it may change.

Reviewer: The sentence would work also without "it", I think.
